events.js

```diff
@@ -1,472 +1,488 @@
+  //back around to the key. this is used for switching away from
+  //peers that stall so that you'll rotate through all the peers
+  //not just swich between two different peers.
+
+  function eachFrom(keys, key, iter) {
+    var i = keys.indexOf(key)
+    if(!~i) return
+    //start at 1 because we want to visit all keys but key.
+    for(var j = 1; j < keys.length; j++)
+      if(iter(keys[(j+i)%keys.length], j))
+        return
+  }
 
-function setNotes (peer, feed, seq, rx) {
-  peer.notes = peer.notes || {}
-  peer.notes[feed] = opts.note(seq, rx)
+  function setNotes (peer, feed, seq, rx) {
+    peer.notes = peer.notes || {}
+    peer.notes[feed] = opts.note(seq, rx)
 
-  var rep = peer.replicating[feed]
-  if(rep) {
-    //note: v2 doesn't have a way to represent seq=0 but don't rx, so always rx if zero.
-    rep.rx = getReceive(peer.notes[feed])
-    rep.requested = seq
+    var rep = peer.replicating[feed]
+    if(rep) {
+      //note: v2 doesn't have a way to represent seq=0 but don't rx, so always rx if zero.
+      rep.rx = getReceive(peer.notes[feed])
+      rep.requested = seq
+    }
   }
-}
 
-exports.initialize = function (id) {
-  return {
-    id: id,
-    clock: null,
-    follows: {},
-    blocks: {},
-    peers: {},
-    receive: []
+  // defaults for backwards compatibility
+  exports.getMsgAuthor = function(msg) {
+    return msg.author
+  }
+  exports.getMsgSequence = function(msg) {
+    return msg.sequence
   }
-}
 
-exports.clock = function (state, clock) {
-  state.clock = clock
-  return state
-}
+  exports.initialize = function (id, getMsgAuthor, getMsgSequence) {
+    if (getMsgAuthor)
+      exports.getMsgAuthor = getMsgAuthor
+    if (getMsgSequence)
+      exports.getMsgSequence = getMsgSequence
+
+    return {
+      id: id,
+      clock: null,
+      follows: {},
+      blocks: {},
+      peers: {},
+      receive: []
+    }
+  }
 
-exports.connect = function (state, ev) {
-  if(state.peers[ev.id]) throw new Error('already connected to peer:'+ev.id)
-  if(typeof ev.client != 'boolean') throw new Error('connect.client must be boolean')
-
-//  if(isBlocked(state, state.id, ev.id)) return state
-
-  state.peers[ev.id] = {
-    blocked: isBlocked(state, state.id, ev.id),
-    clock: null,
-    client: !!ev.client,
-    msgs: [],
-    retrive: [],
-    notes: null,
-    //if we are client, wait until we receive notes to send code.
-    //this is a weird way of doing it! shouldn't we just have a bit of state
-    //for wether we have received a vector clock
-    replicating: ev.client ? null : {}
+  exports.clock = function (state, clock) {
+    state.clock = clock
+    return state
   }
 
-  return state
-}
+  exports.connect = function (state, ev) {
+    if(state.peers[ev.id]) throw new Error('already connected to peer:'+ev.id)
+    if(typeof ev.client != 'boolean') throw new Error('connect.client must be boolean')
+
+    //  if(isBlocked(state, state.id, ev.id)) return state
```

Comment:
commented out without a comment => remove?

Reply:
Agree, that is a bit strange. It was already there, I didn't actually see it before you pointed it out.